### PR TITLE
[Story] Add bulk delete to Label Manager (#444)

### LINKED ISSUE #444: [Story] Bulk delete labels in Label Manager
## Description

Label Manager already supports creating, editing, and deleting a **single** label from the Labels tab (row-level Delete, with a repository-scoped confirmation dialog). Dogfooding showed that cleaning up several leftover labels still means repeating that flow once per row.

This story adds **bulk delete** on the Labels tab: select one or more labels, choose Delete, confirm with an “Are you sure?” dialog, then delete the selected labels from the relevant repositories.

## User Story

**As a** solo developer managing labels across several repositories,
**I want** to select multiple labels in Label Manager and delete them in one confirmed action,
**So that** I can clean up unused or unwanted labels without repeating the per-row delete flow.

## Proposed UX

- Add multi-select on the Labels tab grid (one or more rows).
- Provide a bulk **Delete** action (toolbar or equivalent), enabled only when at least one label is selected and at least one repository is in scope.
- Keep the existing per-row Delete action for a single label.
- On bulk Delete, show a confirmation dialog that asks **Are you sure?** and names the labels that will be deleted (and, if more than one repository is selected, which repositories are in scope).
- **Yes** (or equivalent confirm) deletes those labels; **No** / cancel leaves labels unchanged.
- After a successful or partial run, refresh the grid and report how many labels were deleted (and any per-label or per-repository failures).

## Acceptance Criteria

- [ ] Given one or more repositories are selected and labels are loaded, when I select one or more label rows and choose Delete, then a confirmation dialog asks whether I am sure and lists the labels (and repositories) that will be affected.
- [ ] Given the confirmation dialog is open, when I cancel or choose No, then no labels are deleted and my selection is unchanged.
- [ ] Given I confirm Yes, when the operation runs, then each selected label is deleted from every selected repository that currently has that label.
- [ ] Given a label is missing from some selected repositories, when bulk delete runs, then those repositories are skipped for that label and the remaining targets still proceed.
- [ ] Given a GitHub API error for one label or repository, when bulk delete continues, then remaining items still run and the UI reports a clear per-item error rather than failing silently.
- [ ] Given no rows are selected, when I view the Labels tab, then the bulk Delete action is disabled or unavailable.
- [ ] The user guide for Label Manager and the Playwright coverage for `/labels` describe and exercise bulk delete (confirm, cancel, and a successful delete path or documented docs-capture equivalent).

## Related Epic / Milestone

Follow-on to closed Label Manager work: Feature [#27](https://github.com/markheydon/solo-dev-board/issues/27) and Story [#33](https://github.com/markheydon/solo-dev-board/issues/33) (per-row create / edit / delete across repositories). Not a duplicate of those issues: they delivered single-label CRUD, not multi-select bulk delete on the Labels tab.

Milestone **v1.1.0**.

## Implementation References

- **Wireframe:** [plan/wireframes/label-manager-wireframe.md](https://github.com/markheydon/solo-dev-board/blob/main/plan/wireframes/label-manager-wireframe.md) (Labels-tab bulk delete).
- **Parent issue:** N/A (original Feature [#27](https://github.com/markheydon/solo-dev-board/issues/27) is closed).
- **Test issue:** [#459](https://github.com/markheydon/solo-dev-board/issues/459).
- **Related decisions:** DEC-034.
- **Feature plan doc:** [plan/label-manager-v1.1-dogfood-project-plan.md](https://github.com/markheydon/solo-dev-board/blob/main/plan/label-manager-v1.1-dogfood-project-plan.md).
- **User guide:** [Label Manager](https://solodevboard.com/docs/label-manager/) (`website/content/docs/label-manager.md`).

### Current behaviour (do not treat as already done)

- The Labels tab action strip has **New label** and **Load selected repositories** only; `MudDataGrid` has no multi-select.
- Row Delete opens `LabelOperationDialog` (“Delete label from the selected repositories”) and then calls `ILabelManagerService.DeleteLabelAsync` for **one** label name.
- Recommended taxonomy (optional **Remove labels outside taxonomy**) and Synchronise can already delete many labels via preview/apply. This story is specifically bulk delete from the **Labels** tab grid.

### GitHub API and product constraints

- GitHub REST deletes **one** label per repository per call (`DELETE /repos/{owner}/{repo}/labels/{name}`). There is no bulk-delete endpoint; the app must loop.
- GitHub typically allows deleting a label even when it is still applied to issues or pull requests; those items simply lose the label. Do not invent a “cannot delete while in use” rule unless the API actually returns an error.
- Label Manager CRUD does **not** protect GitHub default or system labels. Recommended taxonomy clean-up also has **no** protected allow-list (see the user guide warning). Bulk delete should match that: any selected label is eligible, with confirmation as the safety net.
- Existing `DeleteLabelAsync` is sequential and currently **stops on the first failure**. Bulk delete should prefer the taxonomy-apply pattern: continue the batch and surface per-label / per-repository errors.
- Consolidated rows may exist in only a subset of selected repositories; delete only where the label is present.
- Disable duplicate submit while a bulk delete is in progress (same idea as Synchronise apply).

Source: dogfooding of Label Manager.

### PR DESCRIPTION
**Pull request by markheydon**
## Summary of Changes

Adds bulk delete on the Label Manager **Labels** tab: multi-select rows in the consolidated grid, confirm via an "Are you sure?" dialog, then delete each selected label from every in-scope repository that currently contains it. The batch continues after per-item GitHub API failures and reports deleted, skipped, and failed counts.

## Related Issue(s)

Closes #444

## Type of Change

- [ ] 🐛 Bug fix (non-breaking change that fixes an issue)
- [x] ✨ Feature (non-breaking change that adds functionality)
- [ ] 🔧 Chore / maintenance (dependency update, refactoring, tooling)
- [ ] ♻️ Refactoring (no functional change, improves code quality)
- [ ] 📝 Documentation (updates to docs, comments, or README)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] I have performed a self-review of my own code
- [x] My changes follow the coding conventions in `AGENTS.md`
- [x] All new and existing tests pass (`dotnet test`)
- [x] I have added or updated tests to cover my changes
- [x] I have updated relevant documentation in `docs/` or `plan/`
- [x] No new compiler warnings have been introduced
- [x] All text in comments, strings, and docs is written in **UK English**
- [x] If this adds a new feature, a GitHub Issue exists (or is updated), Project #8 is synced, and end-user docs under `website/` (or developer docs under `docs/`) have been updated

## Screenshots (if applicable)

No screenshots attached. UI change is covered by component tests (`LabelsTests`) and Playwright shell assertions in `labels.spec.ts`.

## Additional Notes

- **Application:** `BulkDeleteLabelsAsync` loops label-repository targets, skips missing labels (`KeyNotFoundException`), and records per-item errors without stopping the batch (taxonomy-apply pattern).
- **UI:** `MudDataGrid` multi-select with `SelectColumn`, toolbar **Delete** action, confirmation dialog listing labels and repositories, in-progress indicator, and duplicate-submit guard.
- **Tests:** Four `LabelServiceTests`, three `LabelsTests` (including mocked dialog confirm/cancel), and Playwright shell assertions for disabled bulk delete.
- Per-row Delete via `LabelOperationDialog` is unchanged.
- Test issue: #459

<div><a href="https://cursor.com/agents/bc-15ef40ba-f3af-41f7-9e44-cba8867d2294?cursor_ref=pr_footer&cursor_cta=open_in_web"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-web-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-web-light.png">Open in Web</picture></a>&nbsp;<a href="https://cursor.com/background-agent?bcId=bc-15ef40ba-f3af-41f7-9e44-cba8867d2294&cursor_ref=pr_footer&cursor_cta=open_in_cursor"><picture><source media="(prefers-color-scheme: dark)" srcset="https://cursor.com/assets/images/open-in-cursor-dark.png"><source media="(prefers-color-scheme: light)" srcset="https://cursor.com/assets/images/open-in-cursor-light.png">Open in Cursor</picture></a>&nbsp;</div>

### CODE REVIEW

File: src/App/SoloDevBoard.App/Components/Features/Labels/Pages/Labels.razor.cs [hunk 1]
```diff
@@ -836,6 +838,135 @@ private async Task ExecuteDeleteAsync(LabelOperationDialogResult operation)
...
+        isBulkDeletingLabels = true;
+        await InvokeAsync(StateHasChanged);
+
+        try
+        {
+            var result = await LabelManagerService.BulkDeleteLabelsAsync(targets);
+
+            if (!result.HasErrors)
+            {
+                Snackbar.Add($"Deleted {result.DeletedCount} label-repository pairs.", Severity.Success);
```
Comment (markheydon) [inline src/App/SoloDevBoard.App/Components/Features/Labels/Pages/Labels.razor.cs]: **Medium** — `LabelBulkDeleteResultDto.SkippedCount` is never reported in the snackbar. The user guide states that repositories without a label are skipped for that name, but the success path only shows `DeletedCount`. Consider including skipped pairs in the summary (for example, "Deleted 3 label-repository pairs (1 skipped).") so operators can reconcile the outcome with the documented behaviour.

File: src/App/SoloDevBoard.App/Components/Features/Labels/Dialogs/LabelBulkDeleteConfirmDialog.razor [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+            }
+        </MudList>
+
+        <MudText Typo="Typo.subtitle2" Class="mt-3">Repositories in scope</MudText>
+        <MudList T="string" Dense="true" data-testid="bulk-delete-repository-list">
+            @foreach (var repositoryFullName in Content.RepositoryFullNames)
+            {
+                <MudListItem T="string">@repositoryFullName</MudListItem>
+            }
+        </MudList>
```
Comment (markheydon) [inline src/App/SoloDevBoard.App/Components/Features/Labels/Dialogs/LabelBulkDeleteConfirmDialog.razor]: **Low** — The dialog lists every repository in scope, not the per-label subset that will actually be deleted (`RepositoriesWithLabel` ∩ selected repos). That is accurate at batch level but can overstate impact when a label exists in only one of several selected repositories. Optional polish: show per-label affected repositories, or soften the heading to "Repositories in scope (labels are removed only where present)."

File: src/App/SoloDevBoard.App/Components/Features/Labels/Pages/Labels.razor [hunk 3]
```diff
@@ -45,20 +45,34 @@
...
+                                   OnClick="ShowBulkDeleteConfirmDialogAsync">
+                            Delete
+                        </MudButton>
                     </MudStack>
 
+                    @if (IsLabelsTabBusy)
+                    {
+                        <MudProgressLinear Indeterminate="true" Color="Color.Primary" aria-label="@BulkDeleteProgressAriaLabel" data-testid="bulk-delete-progress-indicator" />
+                        <MudText Typo="Typo.body2" aria-live="polite" role="status">@BulkDeleteProgressMessage</MudText>
+                    }
```
Comment (markheydon) [inline src/App/SoloDevBoard.App/Components/Features/Labels/Pages/Labels.razor]: **Low** — Taxonomy apply has component tests for in-progress UI and double-submit prevention (`Labels_ApplyRecommendedTaxonomy_WhenClickedTwiceDuringPendingCall_CallsServiceOnce`). Consider adding equivalent coverage for bulk delete (progress indicator visible, Delete disabled while `isBulkDeletingLabels`) to lock in the duplicate-submission guard.

**Review by markheydon — COMMENTED** [review]
## Code review — PR #461

**Merge recommendation:** Approve with minor concerns

Solid implementation that follows layered architecture, MudBlazor patterns, and the taxonomy-apply batch-error model. Tests and user-guide alignment are in good shape.

### Summary

| Severity | Count | Theme |
|----------|-------|-------|
| Critical | 0 | — |
| High | 0 | — |
| Medium | 1 | `SkippedCount` not surfaced in snackbars |
| Low | 2 | Confirm dialog scope wording; missing in-progress component test parity |

### Strengths

- Application service returns DTOs; bulk delete logic lives in `LabelService` with `ConfigureAwait(false)` and per-item error continuation.
- UI mirrors taxonomy apply: progress indicator, duplicate-submit guard, auth recovery, and grid reload in `finally`.
- Component tests cover disabled state, confirm, and cancel; application tests cover success, skip, partial failure, and empty targets.
- User guide, `USER_DOCS_ALIGNMENT.md`, `CRITICAL_JOURNEYS.md`, and Playwright shell assertions updated.
- PR title, template body, labels, and `Closes #444` linkage match `PULL_REQUEST_POLICY.md`.

### Verified

- `dotnet test` — all seven `BulkDelete`-filtered tests passed.

Inline threads below are actionable for Delivery; none block merge from an architecture or correctness standpoint.

File: src/App/SoloDevBoard.App/Components/Features/Labels/Pages/Labels.razor.cs (hunk 1 above)
Comment (markheydon) [reply src/App/SoloDevBoard.App/Components/Features/Labels/Pages/Labels.razor.cs]: Addressed in 0876a52 — see commit on this branch.

File: src/App/SoloDevBoard.App/Components/Features/Labels/Dialogs/LabelBulkDeleteConfirmDialog.razor (hunk 2 above)
Comment (markheydon) [reply src/App/SoloDevBoard.App/Components/Features/Labels/Dialogs/LabelBulkDeleteConfirmDialog.razor]: Addressed in 0876a52 — see commit on this branch.

File: src/App/SoloDevBoard.App/Components/Features/Labels/Pages/Labels.razor (hunk 3 above)
Comment (markheydon) [reply src/App/SoloDevBoard.App/Components/Features/Labels/Pages/Labels.razor]: Addressed in 0876a52 — see commit on this branch.

**Comment by markheydon** [comment]
## Address PR review comments — summary

All three review threads on this PR have been implemented, replied to, and resolved in commit `0876a52`.

### Changes

1. **Skipped count in snackbars (Medium)** — `FormatBulkDeleteSummaryMessage` now includes skipped label-repository pairs when `SkippedCount > 0` (for example, `Deleted 2 label-repository pairs (1 skipped).`).
2. **Confirmation dialog scope (Low)** — Dialog payload now lists each label with its affected repositories (`LabelBulkDeleteConfirmDialogLabelTarget`). Body copy clarifies labels are removed only where present.
3. **In-progress and duplicate-submit tests (Low)** — Added component tests for progress UI, disabled Delete while busy, double-click prevention (`isAwaitingBulkDeleteConfirmation`), and skipped-count snackbar text. Six `BulkDelete`-filtered tests pass locally.

### Docs

- Updated `website/content/docs/label-manager.md` to describe per-label repository listing in the confirmation dialog.

Ready for re-review.
